### ISSUE TITLE
Is there a plan to port this wonderful control to Firemonkey?

### ISSUE THREAD

**Issue by GaNacereddine** [digao-dalpiaz/DzHTMLText#34]
can this control be ported to firemonkey?

**Comment by digao-dalpiaz** [comment]
Hello @GaNacereddine 

Yes, there is! I'm preparing to do this, but takes a lot of time because there are many changes in FMX scenario.

Thanks and let's keep on touch.

**Comment by GaNacereddine** [comment]
thank you very much. will be looking forward to that :).

**Comment by HemulGM** [comment]
Connect to waiting)

**Comment by digao-dalpiaz** [comment]
Guys, great news: I'm almost done with FMX support.

![fmx_example](https://user-images.githubusercontent.com/46692097/105937170-c46cdf00-6033-11eb-990e-9192dd64baf1.png)

**Comment by GaNacereddine** [comment]
What a great news to hear, I will be waiting to test. Thank you very much for your valuable time and experience.

**Comment by digao-dalpiaz** [comment]
**Problems to solve:**
- [x] Directive to define platform
- [x] Use OnFontChanged in VCL and remove CMFontChanged (not possible)
- [x] Review FMX default published properties
- [x] TDHVisualItem_ImageResource.Picture
- [x] TDHVisualItem_ImageResource.Load (PNG)
- [x] OnPaint: TDHVisualItem_ImageResource
- [x] ParamToColor function
- [x] Tag_Color_ProcValue
- [x] OnPaint: TDHVisualItem_Image
- [x] Review Cursor Changed behavior (can't use CMCursorchanged in FMX)
- [x] Click: ShellExecute and others platforms
- [x] DoTextAndRelated: Kind ttImage

**Changed behaviors (to write in README):**
- [x] Now color notation supports # or $ with 6 or 8 digits
- [x] Now supports transparent background using clNone (VCL) or Null (FMX) color
- [x] Now supports any image format in Resource when using FMX
- [x] When Delphi below XE8, there is no ImageList capability
- [x] Using only first ImageList Layer for now
- [x] Others platforms than Windows and Linux, like Android, iOS and MacOS doesn't open links directly

**To do list:**
- [x] Add VCL prefix in VCL unit
- [x] Old FMX Delphi have different FMX uses (needs directive)
- [x] Change version to 3.0
- [x] Readme change log info
- [x] Readme review
- [ ] Missing Alpha when using transparent background in old Delphis using FMX

**Testing:**
- [ ] Delphi XE2
- [x] Delphi XE3
- [ ] Delphi XE4
- [x] Delphi XE5
- [x] Delphi XE6
- [ ] Delphi XE7
- [ ] Delphi XE8
- [ ] Delphi 10 Seattle
- [ ] Delphi 10.1 Berlin
- [ ] Delphi 10.2 Tokyo
- [x] Delphi 10.3 Rio
- [x] Delphi 10.4 Sydney

**Final actions:**
- [ ] Release DzHTMLText 3.0 branch
- [ ] Release Dam 4.9 branch

**Comment by digao-dalpiaz** [comment]
There is a release candidate version available in this branch: https://github.com/digao-dalpiaz/DzHTMLText/tree/fmx-support

Please let me know if you are testing on a version of Delphi that I have not yet tested according to the previous list.

**Comment by GaNacereddine** [comment]
Unfortunately, I only own 10.3 and 10.4. the you already tested.
I used to have 10.2 but I lost the license.

**Comment by digao-dalpiaz** [comment]
> Unfortunately, I only own 10.3 and 10.4. the ones you already tested.
> I used to have 10.2 but I lost the license.

Even so, if you can, do tests on the versions you have to see if the features are as expected.

**Comment by GaNacereddine** [comment]
Okay will definitely do that I have a small project where I could test this control in the real world.

**Comment by digao-dalpiaz** [comment]
I finished a last commit just now. I believe everything is already ok, just missing to make it work in XE2.

*The project list has been updated in the comment above.*

**Comment by digao-dalpiaz** [comment]
Hello @GaNacereddine and @HemulGM,

Could you run tests on new version, please?
Download link: https://github.com/digao-dalpiaz/DzHTMLText/tree/fmx-support

Readme info already updated and Component Installer for Auto Installation already updated too.

Thanks a lot.

**Comment by HemulGM** [comment]
Sure. I'll be glad to help)

**Comment by HemulGM** [comment]
FMX Windows - isOk
FMX Android - is not working (There is no formatting. Just text. I'll try to figure out when the time will be)

**Comment by GaNacereddine** [comment]
Sure I'm setting up 10.4.1 on my dev pc, I will update you once the setup is done (damn patches :) )

**Comment by digao-dalpiaz** [comment]
> FMX Windows - isOk
> FMX Android - is not working (There is no formatting. Just text. I'll try to figure out when the time will be)

I discovered the problem on Android. I'm going to upload adjustments right now.

**Comment by HemulGM** [comment]
FMX Android - isOk, but I found a bug on android. Its nature is clear - there is no MouseMove event.

Click on the spoiler and then just click anywhere. A spoiler click will be triggered.

**Comment by HemulGM** [comment]
Create new issues or can I write them here?

**Comment by digao-dalpiaz** [comment]
> Create new issues or can I write them here?

You can create a new issue. Then I'll tag it with FMX support reference.
Thanks

**Comment by digao-dalpiaz** [comment]
Tests ok also on:
- Lazarus 2.0.10 in Windows environment
- Lazarus 2.0.10 in Linux environment (Debian/Ubuntu/Linux Lite)
